src/OpenTelemetry.Extensions/Trace/BaggageActivityProcessor.cs

```diff
@@ -1,21 +1,41 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+using System;
 using System.Diagnostics;
 
 namespace OpenTelemetry.Trace;
 
 /// <summary>
-/// Activity processor that adds <see cref="Baggage"/> fields to every new span.
+/// Activity processor that adds <see cref="Baggage"/> fields to every new activity.
 /// </summary>
 internal sealed class BaggageActivityProcessor : BaseProcessor<Activity>
 {
+    /// <summary>
+    /// A predicate that allows all baggage keys.
+    /// </summary>
+    public static readonly Predicate<string> AllowAllBaggageKeys = _ => true;
+
+    private readonly Predicate<string> baggageKeyPredicate;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BaggageActivityProcessor"/> class.
+    /// </summary>
+    /// <param name="baggageKeyPredicate">Predicate to determine which baggage keys should be added to the activity.</param>
+    public BaggageActivityProcessor(Predicate<string> baggageKeyPredicate)
+    {
+        this.baggageKeyPredicate = baggageKeyPredicate ?? throw new ArgumentNullException(nameof(baggageKeyPredicate));
+    }
+
     /// <inheritdoc />
     public override void OnStart(Activity activity)
     {
         foreach (var entry in Baggage.Current)
         {
-            activity.SetTag(entry.Key, entry.Value);
+            if (this.baggageKeyPredicate(entry.Key))
```

Comment:
I am not sure if running arbitrary user code run here is recommended.
Within OTel SDK, regex pattern was used instead:
Metrics : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MeterProviderSdk.cs#L154-L155 (allowing user to decide which meter names are enabled). Similar for Traces too.

Could we just accept a Regex pattern instead?

Reply:
I understand the concerns of running user-provided code. I'm trying super hard to be consistent across SDK implementations. Some want user-function, some want list of prefix strings and others want a regex. The user-function supports the other two.

Would you be okay keeping the user function if we add a regex extension method that takes a regex directly and add a warning to providing custom functions around being aware they are executed per activity?

Reply:
> custom functions around being aware they are executed per activity

It is per activity, for *each* baggage entry.

Reply:
I totally understand you are trying to get as much consistency as possible across languages, but the only sure way to make that happen is to have it come via Spec.

I'd prefer to *only* offer the Regex option here, it is consistent with how OTel .NET allows meter/activity-source. However, If other maintainers want to keep both, I won't block this PR!

Reply:
@open-telemetry/dotnet-contrib-maintainers Please share your thoughts.

Reply:
I'm not aware of spec that covers contrib components. If you can point me towards one I'd be happy to lobby it.

What are your concerns regarding providing the user-function? Security, performance, something else? In my mind, this is an opt-in processor that requires some level of configuration to select which baggage keys are selected. A regex is one way someone may want to configure it, but may not suit everyone.

I'm not dead-set on it being there, it's a goal to make the developer experience as consistent as possible. If it's a no-go from this SIG, I can deal with it.

Reply:
BTW we already use a user-function in the [AutoFlushActivityProcessor](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/Trace/AutoFlushActivityProcessor.cs#L41). It has instructions that function should not be complex (time-consuming) - would adding that warning be sufficient?

Reply:
That's my point, this is being added to contrib because we don't have a spec for it. My goal is to keep them consistent for developer experience as best as possible.

I'm not hard-set for it, but there is precedent for other contrib processors using a user-provided function.